api/src/main/java/org/apache/iceberg/actions/Cdc.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+public interface Cdc extends Action<Cdc, Cdc.Result> {
+  /**
+   * Emit changed data set by a snapshot id.
+   *
+   * @param snapshotId id of the snapshot to generate changed data
+   * @return this for method chaining
+   */
+  Cdc ofSnapshot(long snapshotId);
+
+  /**
+   * Emit changed data set by the current snapshot.
+   *
+   * @return this for method chaining
+   */
+  Cdc ofCurrentSnapshot();
```

Comment:
In which case will this method be useful? I guess consumers will probably always want to consume from a particular point in time so I am not sure CDC records for a current snapshot would be very useful.

Reply:
+1. for incremental read, we need to bookkeep the position. implicit current snapshot would make the bookkeeping impossible and hence is probably not a valid use case